SignalMessaging/utils/ConversationSearcher.swift

```diff
@@ -75,30 +111,44 @@ public class ConversationSearcher: NSObject {
             if let thread = match as? TSThread {
                 let threadViewModel = ThreadViewModel(thread: thread, transaction: transaction)
                 let snippet: String? = thread.lastMessageText(transaction: transaction)
-                let searchResult = ConversationSearchResult(thread: threadViewModel, messageId: nil, snippet: snippet)
+                let sortKey = NSDate.ows_millisecondsSince1970(for: threadViewModel.lastMessageDate)
+                let searchResult = ConversationSearchResult(thread: threadViewModel, messageId: nil, snippet: snippet, sortKey: sortKey)
 
                 if let contactThread = thread as? TSContactThread {
                     let recipientId = contactThread.contactIdentifier()
                     existingConversationRecipientIds.insert(recipientId)
                 }
+
                 conversations.append(searchResult)
             } else if let message = match as? TSMessage {
                 let thread = message.thread(with: transaction)
 
                 let threadViewModel = ThreadViewModel(thread: thread, transaction: transaction)
-                let searchResult = ConversationSearchResult(thread: threadViewModel, messageId: message.uniqueId, snippet: snippet)
-
+                let sortKey = message.timestamp
+                let searchResult = ConversationSearchResult(thread: threadViewModel, messageId: message.uniqueId, snippet: snippet, sortKey: sortKey)
                 messages.append(searchResult)
             } else if let signalAccount = match as? SignalAccount {
-                let searchResult = ContactSearchResult(signalAccount: signalAccount)
+                let anyFont = UIFont.systemFont(ofSize: 12)
+                let sortKey = contactsManager.attributedStringForConversationTitle(withPhoneIdentifier: signalAccount.recipientId,
```

Comment:
How does this work? It looks like we'll sort by display order (e.g. by first name in most english settings). 

Instead we should respect the "sort by first name vs. last name" setting so it's the same order as what we see in the contact picker.

Reply:
Good point, done.